src/util/util.c

```diff
@@ -589,6 +589,14 @@ errno_t sss_filter_sanitize_ex(TALLOC_CTX *mem_ctx,
             output[j++] = '5';
             output[j++] = 'c';
             break;
+        case '\n':
+            output[j++] = '\\';
+            output[j++] = '0';
+            output[j++] = 'd';
+            output[j++] = '\\';
```

Comment:
Just ran a test with \r and it doesn't trigger the removal of a valid user (ad1 in my case):

ubuntu@vtemp-cache-x:~⟫ sudo sss_cache -E; getent passwd $(echo -e "ad1\rbad")

Log:
(Mon Mar  6 10:19:28 2017) [sssd[be[UBUNTU.TEST]]] [sdap_get_generic_ext_step] (0x0400): calling ldap_search_ext with [(&(sAMAccountName=ad1^Mbad)(objectclass=user)(sAMAccountName=*)(&(uidNumber=*)(!(uidNumber=0))))][CN=Users,DC=ubuntu,DC=test].
...
(Mon Mar  6 10:19:28 2017) [sssd[be[UBUNTU.TEST]]] [sdap_search_user_process] (0x0400): Search for users, returned 0 results.
(Mon Mar  6 10:19:28 2017) [sssd[be[UBUNTU.TEST]]] [sdap_get_users_done] (0x0040): Failed to retrieve users
...
(Mon Mar  6 10:19:28 2017) [sssd[be[UBUNTU.TEST]]] [sysdb_delete_user] (0x0400): Error: 2 (No such file or directory)

Shall we sanitize \r anyway?